### PR TITLE
[GLUTEN-6900] Use clang-format-18

### PR DESCRIPTION
**Pull request by xumingming**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#6900)

### CODE REVIEW

**Comment by FelixYBW** [comment]
Why do we need to upgrade clang-format?

**Comment by xumingming** [comment]
@FelixYBW The reason is as follows:

1. When I change some c++ code, and submit the PR, the PR will fail at code style check.
2. I noticed there is `dev/formatcppcode.sh`, but it only support Linux
3. After make `dev/formatcppcode.sh` support macOS, I found that clang-format on macOS(which is of version 18) behaves differently from clang-format-15 on Linux.

Do you have suggestions?

File: dev/formatcppcode.sh [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+else
+    CLANG_FORMAT_CMD="clang-format-18"
+    if ! command -v $CLANG_FORMAT_CMD &> /dev/null
+    then
+        echo "$CLANG_FORMAT_CMD could not be found"
+        echo "Installing $CLANG_FORMAT_CMD..."
+        sudo apt update
+        sudo apt install $CLANG_FORMAT_CMD
+    fi
 fi
```
Comment (FelixYBW) [inline dev/formatcppcode.sh:21]: Let's remove the install commands here. It's only for darwin and ubuntu.

**Comment by FelixYBW** [comment]
> @FelixYBW The reason is as follows:
> 
> 1. When I change some c++ code, and submit the PR, the PR will fail at code style check.
> 2. I noticed there is `dev/formatcppcode.sh`, but it only support Linux
> 3. After make `dev/formatcppcode.sh` support macOS, I found that clang-format on macOS(which is of version 18) behaves differently from clang-format-15 on Linux.
> 4. And there seems no easy way to install clang-format-15 on macOS.
> 
> Do you have suggestions?

Different OS has different default clang-format version. e.g. I'm using Ubuntu20.04 which has clang-format-12. Do 12 and 18 have any difference on format? format-12 works well on my side so far.

**Comment by xumingming** [comment]
> > @FelixYBW The reason is as follows:
> > 
> > 1. When I change some c++ code, and submit the PR, the PR will fail at code style check.
> > 2. I noticed there is `dev/formatcppcode.sh`, but it only support Linux
> > 3. After make `dev/formatcppcode.sh` support macOS, I found that clang-format on macOS(which is of version 18) behaves differently from clang-format-15 on Linux.
> > 4. And there seems no easy way to install clang-format-15 on macOS.
> > 
> > Do you have suggestions?
> 
> Different OS has different default clang-format version. e.g. I'm using Ubuntu20.04 which has clang-format-12. Do 12 and 18 have any difference on format? format-12 works well on my side so far.

Yes, they have many difference. Seems not easy to get a same output on macOS, closing it.

**Pull request closed without merging**